packages/api/src/store/queue.ts

```diff
@@ -91,6 +99,8 @@ export class RabbitQueue implements Queue {
           channel.bindQueue(QUEUES.webhooks, EXCHANGES.webhooks, "webhooks.#"),
           channel
             .assertQueue(QUEUES.delayed, {
+              // Quorum queues do not support message expiration, so this has to
```

Comment:
Just a thing I am missing, does this affect also the events and webhooks queues? What happens if the consumers on these two queues fails to consume the messages and they haven't a TTL?

Reply:
This means that we won't be able to send messages with an expiration to the events/webhooks queues. Which at least currently is the case, since we don't want messages to disappear directly within RabbitMQ, but want to explicitly handle the messages that we can't deliver instead, after the specified number of retries and emailing the user, in the case of webhooks.

If we ever want to have messages that have some expiration and thus are not worth processing after some period, we could either set that expiration inside the payload and just immediately ignore/ack those messages when processing, or create a separate queue which is not of `quorum` type to be able to set an `expiration` for them within RabbitMQ.

In the case of the delayed messages, the expiration is actually only the delay that the message spends in the delayed queue before being dead-letter published to the regular queues. You can check this comment for a slightly longer explanation: https://github.com/livepeer/livepeer-com/blob/d2c7b91ca7dfb5fb9eeea201638066d53843018c/packages/api/src/store/queue.ts#L172